src/backend/pluginLoaders.test.js

```diff
@@ -271,6 +289,35 @@ describe("src/backend/pluginLoaders", () => {
     });
   });
 
+  describe("createReferenceDetector", () => {
+    it("should create a CascadingReferenceDetector", async () => {
+      // Given
+      const loaders = mockPluginLoaders();
+      const cache = mockCacheProvider();
+      const githubToken = exampleGithubToken;
+      const project = createProject({
+        id: "has-github-and-discourse",
+        discourseServer: {serverUrl: "http://foo.bar"},
+        repoIds: [exampleRepoId],
+      });
+      const cachedProject = ({project, cache}: any);
+
+      // When
+      const references = await PluginLoaders.createReferenceDetector(
+        loaders,
+        {githubToken},
+        cachedProject
+      );
+
+      // Then
+      expect(references).toBeInstanceOf(CascadingReferenceDetector);
+      expect(((references: any): CascadingReferenceDetector).refs).toEqual([
+        fakes.githubReferences,
+        fakes.discourseReferences,
+      ]);
```

Comment:
aside: i'm skeptical that tests like this add any value. given that we aren't using the underlying logic, just essentially checking that pieces were piped around correctly, what does this get us that flow doesn't?

Reply:
Flow would not validate whether we've correctly interpreted the input Project to compose the detectors. Because these are all valid type-wise, but only the last is correct logic-wise (given the Project of this test):
- `new CascadingReferenceDetector([])`
- `new CascadingReferenceDetector([githubRefs])`
- `new CascadingReferenceDetector([discourseRefs])`
- `new CascadingReferenceDetector([githubRefs, discourseRefs])`

Will agree that it's not a durable test, but a smoke test is better than no test. As it would not cover any new plugins with reference detectors as they're added. Happy to improve if you have suggestions.